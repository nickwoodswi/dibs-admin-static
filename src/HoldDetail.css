.hold-detail{
    width: 100%;
    max-width: 600px;
    min-width: 300px;
    color: white;
    display: flex;
    flex-wrap: wrap;
    margin: 5px;
    flex-direction: column;
    border-radius: 5px;
    background-color: rgb(0, 0, 0, .9);
    box-shadow: -2px 2px 5px rgb(31, 31, 31);
}
    .hold-detail-header{
        width: 100%;
        display: flex;
        flex-wrap: wrap;
        margin: 0px;
    }
        .hold-date{
            width: 100%;
            background-color: white;
            color: black;
            padding: 5px;
            border-top-left-radius: 5px;
            border-top-right-radius: 5px;
        }
        .event-title-contact{
            width: 100%;
            display: flex;
            flex-direction: column;
        }
            .event-title-status{
                width: 100%;
                display: flex;
                flex-wrap: nowrap;
            }
                .title-support{
                    width: 100%;
                    display: flex;
                    flex-wrap: wrap;
                }
                    .event-title{
                        width: 100%;
                        display: flex;
                        align-items: center;
                        padding: 5px;
                        font-size: 24px;
                    }
                    .event-support{
                        width: 100%;
                        padding: 5px;
                    }
                .order-status{
                    width: 110px;
                    display: flex;
                    flex-direction: column;
                }
                    .hold-order{
                        height: 80px;
                        width: 100%;
                        background-color: red;
                        display: flex;
                        justify-content: center;
                        align-items: center;
                        margin-top: 2px;
                    }
                    .new{
                        width: 100%;
                        background-color: white;
                        display: flex;
                        align-items: center;
                        justify-content: center;
                        color: black;
                        margin-top: 2px;
                        margin-bottom: 2px;
                    }
                        .held{
                            width: 100%;
                            background-color: yellow;
                            display: flex;
                            align-items: center;
                            justify-content: center;
                            color: black;
                            margin-top: 2px;
                            margin-bottom: 2px;
                        }
                        .challenged{
                            width: 100%;
                            background-color: orange;
                            display: flex;
                            align-items: center;
                            justify-content: center;
                            color: white;
                            margin-top: 2px;
                            margin-bottom: 2px;
                        }
                        .offered{
                            width: 100%;
                            background-color: blue;
                            display: flex;
                            align-items: center;
                            justify-content: center;
                            color: white;
                            margin-top: 2px;
                            margin-bottom: 2px;
                        }
                        .confirmed{
                            width: 100%;
                            background-color: green;
                            display: flex;
                            align-items: center;
                            justify-content: center;
                            color: white;
                            margin-top: 2px;
                            margin-bottom: 2px;
                        }
                        .released{
                            width: 100%;
                            background-color: red;
                            display: flex;
                            align-items: center;
                            justify-content: center;
                            color: white;
                            margin-top: 2px;
                            margin-bottom: 2px;
                        }
            .event-contact{
                width: 100%;
                display: flex;
                border-top: 1px dotted red;
                border-bottom: 1px dotted red;
                height: 110px;
            }
                .name-category-organization{
                    width: 50%;
                    padding: 5px;
                    background-color: white;
                    color: black;
                }
                    .contact-name{
                        width: 100%
                    }
                    .contact-category{
                        width: 100%
                    }
                    .organization{
                        width: 100%
                    }
                .email-phone-address-org{
                    width: 50%;
                    padding: 5px;
                    color: white;
                    border-left: 5px solid red
                }
                    .email{
                        width: 100%;
                    }
                    .phone{
                        width: 100%;
                    }
                    .address{
                        width: 100%;
                    }
                        .address1{
                            width: 100%
                        }
                        .address2{
                            width: 100%
                        }
                        .city-zip{
                            width: 100%
                        }
    .status-button-container{
        border-radius: 3px;
        padding: 5px;
        padding-top: 10px;
        border-top: 1px dotted red;
    }
        .status-button{
            width: 110px;
            height: 50px;
            margin-right: 5px;
            margin-bottom: 5px;
            background-color: white;
            border: none;
            font-family: 'Helvetica', sans-serif
        }
            .status-button:active{
                background-color: grey
            }
    .hold-notes-container{
        height: 100px;
        padding: 5px;
        display: flex;
        overflow-y: scroll;
        background-color: black;
    }
        .hold-notes{
        }
    .music-icons-container{
        background-color: black;
        display: flex;
        padding: 5px;
        border-bottom-left-radius: 5px;
        border-bottom-right-radius: 5px;
    }
        img{
            height: 25px;
            margin-right: 10px;
        }
            img:hover{
                opacity: 50%;
            }
        a {
            color: white
        }