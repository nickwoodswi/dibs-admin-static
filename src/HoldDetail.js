import React, { Component } from 'react'
import './HoldDetail.css'

class HoldDetail extends Component {
    static defaultProps = {
        holdState: {}
    }
    render() {
        return(
            <div className="hold-detail">

                <div className="hold-detail-header">
                    <div className="hold-date">{this.props.holdState.hold_date}</div>
                    <div className="event-title-contact">
                        <div className="event-title-status">
                            <div className="title-support">
                                <div className="event-title">{this.props.holdState.contact_act}</div>
                                <div className="event-support">{this.props.holdState.dsupport_title} / {this.props.holdState.opener_title} / {this.props.holdState.opener2_title}</div>
                            </div>
                            <div className="order-status">
                                <div className="hold-order">{this.props.holdState.hold_number}H</div>
                                <div id="challenged" className={this.props.holdState.hold_status}>{this.props.holdState.hold_status}</div>
                            </div>
                        </div>
                        <div className="event-contact">
                            <div className="name-category-organization">
                                <div className="contact-name">{this.props.holdState.hold_firstname} {this.props.holdState.hold_lastname}</div>
                                <div className="contact-category">{this.props.holdState.contact_type}</div>
                                <div className="organization">{this.props.holdState.contact_org}</div>
                            </div>
                            <div className="email-phone-address-org">
                                <div className="email">{this.props.holdState.hold_email}</div>
                                <div className="phone">{this.props.holdState.contact_phone}</div>
                                <div className="address">
                                    <div className="address-1">{this.props.holdState.org_street1}</div>
                                    <div className="address-2">{this.props.holdState.org_street2}</div>
                                    <div className="city-zip">{this.props.holdState.org_city} {this.props.holdState.org_zip}</div>
                                </div>
                            </div>
                        </div>
                    </div>
                </div>
                <div className="hold-notes-container">
                    <div className="hold-notes">
                        {this.props.holdState.event_detail}
                    </div>
                </div>
                <div className="status-button-container">

                    <a href=
                        {'mailto:' 
                        + this.props.holdState.hold_email 
                        + '?subject=Hold%20for%20' 
                        + this.props.holdState.contact_act
                        + '%20has%20been%20confirmed&body=Hi%20' 
                        + this.props.holdState.hold_firstname
                        + ',%0D%0A%0D%0AThis%20is%20a%20courtesy%20note%20to%20let%20you%20know%20your%20'
                        + this.props.holdState.hold_number
                        + 'H%20on%20'
                        + this.props.holdState.hold_date
                        + '%20for%20'
                        + this.props.holdState.contact_act
                        + '%20has%20been%20confirmed.'}>
                        <button 
                            id="held" 
                            className="status-button" 
                            onClick={() => this.props.changeStatus('held', this.props.id)}>HOLD
                        </button>
                    </a>

                    <a href=
                        {'mailto:' 
                        + this.props.holdState.hold_email 
                        + '?subject=Hold%20for%20' 
                        + this.props.holdState.contact_act
                        + '%20has%20been%20challenged&body=Hi%20' 
                        + this.props.holdState.hold_firstname
                        + ',%0D%0A%0D%0AThis%20is%20a%20courtesy%20note%20to%20let%20you%20know%20your%20'
                        + this.props.holdState.hold_number
                        + 'H%20on%20'
                        + this.props.holdState.hold_date
                        + '%20for%20'
                        + this.props.holdState.contact_act
                        + '%20has%20been%20challenged.'}>
                        <button 
                            id="challenged" 
                            className="status-button" 
                            onClick={() => this.props.changeStatus('challenged', this.props.id)}>CHALLENGE
                        </button>
                    </a>

                    <a href=
                        {'mailto:' 
                        + this.props.holdState.hold_email 
                        + '?subject=Offer%20attached%20for%20' 
                        + this.props.holdState.contact_act
                        + '%20on%20'
                        + this.props.holdState.hold_date
                        + '&body=Hi%20' 
                        + this.props.holdState.hold_firstname
                        + ',%0D%0A%0D%0AI%20have%20attached%20an%20offer%20for%20'
                        + this.props.holdState.contact_act
                        + '%20on%20'
                        + this.props.holdState.hold_date
                        + '.%0D%0A%0D%0APlease%20reply%20within%2048%20hours%20to%20confirm,%20counter,%20or%20release.'}>
                        <button 
                            id="offered" 
                            className="status-button" 
                            onClick={() => this.props.changeStatus('offered', this.props.id)}>OFFER
                        </button>
                    </a>

                    <a href=
                        {'mailto:' 
                        + this.props.holdState.hold_email 
                        + '?subject=Confirming%20' 
                        + this.props.holdState.contact_act
                        + '%20for%20'
                        + this.props.holdState.hold_date
                        + '&body=Hi%20' 
                        + this.props.holdState.hold_firstname
                        + ',%0D%0A%0D%0A'
                        + this.props.holdState.contact_act
                        + '%20has%20been%20confirmed%20for%20'
                        + this.props.holdState.hold_date
                        + '.%20If%20you%20have%20any%20concerns%20or%20questions,%20please%20contact%20us%20immediately.'}>
                        <button 
                            id="confirmed" 
                            className="status-button" 
                            onClick={() => this.props.changeStatus('confirmed', this.props.id)}>CONFIRM
                        </button>
                    </a>

                    <a href=
                        {'mailto:' 
                        + this.props.holdState.hold_email 
                        + '?subject=Hold%20for%20' 
                        + this.props.holdState.contact_act
                        + '%20has%20been%20released&body=Hi%20' 
                        + this.props.holdState.hold_firstname
                        + ',%0D%0A%0D%0AThis%20is%20a%20courtesy%20note%20to%20let%20you%20know%20your%20'
                        + this.props.holdState.hold_number
                        + 'H%20on%20'
                        + this.props.holdState.hold_date
                        + '%20for%20'
                        + this.props.holdState.contact_act
                        + '%20has%20been%20released.'}>
                        <button 
                            id="released" 
                            className="status-button" 
                            onClick={() => this.props.changeStatus('released', this.props.id)}>RELEASE
                        </button>
                    </a>
                </div>
                <div className="music-icons-container">
                    <a href={this.props.holdState.spotify}><div id="spotify" className="music-logo"><img src={require('./images/spotify-logo.png')} /></div></a>
                    <a href={this.props.holdState.youtube}><div id="youtube" className="music-logo"><img src={require("./images/youtube-logo.png")} /></div></a>
                    <a href={this.props.holdState.facebook}><div id="facebook" className="music-logo"><img src={require("./images/facebook-logo.png")} /></div></a>
                    <a href={this.props.holdState.bandcamp}><div id="bandcamp" className="music-logo"><img src={require("./images/bandcamp-logo.png")} /></div></a>
                    <a href={this.props.holdState.website}><div id="website" className="music-logo">SITE</div></a>
                </div>
            </div>
        )
    }
}

export default HoldDetail