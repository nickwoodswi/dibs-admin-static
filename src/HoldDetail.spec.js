import React from 'react';
import { render } from '@testing-library/react';
import HoldDetail from './HoldDetail';
import {MemoryRouter} from 'react-router-dom'
import renderer from 'react-test-renderer'

test('renders learn react link', () => {
  render(<MemoryRouter><HoldDetail /></MemoryRouter>);
});

it('renders the UI as expected', () => {
  const tree = renderer
      .create(<MemoryRouter><HoldDetail /></MemoryRouter>)
      .toJSON();
  expect(tree).toMatchSnapshot();
});
