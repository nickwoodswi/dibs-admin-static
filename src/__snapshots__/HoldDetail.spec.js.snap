// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI as expected 1`] = `
<div
  className="hold-detail"
>
  <div
    className="hold-detail-header"
  >
    <div
      className="hold-date"
    />
    <div
      className="event-title-contact"
    >
      <div
        className="event-title-status"
      >
        <div
          className="title-support"
        >
          <div
            className="event-title"
          />
          <div
            className="event-support"
          >
             / 
             / 
          </div>
        </div>
        <div
          className="order-status"
        >
          <div
            className="hold-order"
          >
            H
          </div>
          <div
            id="challenged"
          />
        </div>
      </div>
      <div
        className="event-contact"
      >
        <div
          className="name-category-organization"
        >
          <div
            className="contact-name"
          >
             
          </div>
          <div
            className="contact-category"
          />
          <div
            className="organization"
          />
        </div>
        <div
          className="email-phone-address-org"
        >
          <div
            className="email"
          />
          <div
            className="phone"
          />
          <div
            className="address"
          >
            <div
              className="address-1"
            />
            <div
              className="address-2"
            />
            <div
              className="city-zip"
            >
               
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="hold-notes-container"
  >
    <div
      className="hold-notes"
    />
  </div>
  <div
    className="status-button-container"
  >
    <a
      href="mailto:undefined?subject=Hold%20for%20undefined%20has%20been%20confirmed&body=Hi%20undefined,%0D%0A%0D%0AThis%20is%20a%20courtesy%20note%20to%20let%20you%20know%20your%20undefinedH%20on%20undefined%20for%20undefined%20has%20been%20confirmed."
    >
      <button
        className="status-button"
        id="held"
        onClick={[Function]}
      >
        HOLD
      </button>
    </a>
    <a
      href="mailto:undefined?subject=Hold%20for%20undefined%20has%20been%20challenged&body=Hi%20undefined,%0D%0A%0D%0AThis%20is%20a%20courtesy%20note%20to%20let%20you%20know%20your%20undefinedH%20on%20undefined%20for%20undefined%20has%20been%20challenged."
    >
      <button
        className="status-button"
        id="challenged"
        onClick={[Function]}
      >
        CHALLENGE
      </button>
    </a>
    <a
      href="mailto:undefined?subject=Offer%20attached%20for%20undefined%20on%20undefined&body=Hi%20undefined,%0D%0A%0D%0AI%20have%20attached%20an%20offer%20for%20undefined%20on%20undefined.%0D%0A%0D%0APlease%20reply%20within%2048%20hours%20to%20confirm,%20counter,%20or%20release."
    >
      <button
        className="status-button"
        id="offered"
        onClick={[Function]}
      >
        OFFER
      </button>
    </a>
    <a
      href="mailto:undefined?subject=Confirming%20undefined%20for%20undefined&body=Hi%20undefined,%0D%0A%0D%0Aundefined%20has%20been%20confirmed%20for%20undefined.%20If%20you%20have%20any%20concerns%20or%20questions,%20please%20contact%20us%20immediately."
    >
      <button
        className="status-button"
        id="confirmed"
        onClick={[Function]}
      >
        CONFIRM
      </button>
    </a>
    <a
      href="mailto:undefined?subject=Hold%20for%20undefined%20has%20been%20released&body=Hi%20undefined,%0D%0A%0D%0AThis%20is%20a%20courtesy%20note%20to%20let%20you%20know%20your%20undefinedH%20on%20undefined%20for%20undefined%20has%20been%20released."
    >
      <button
        className="status-button"
        id="released"
        onClick={[Function]}
      >
        RELEASE
      </button>
    </a>
  </div>
  <div
    className="music-icons-container"
  >
    <a>
      <div
        className="music-logo"
        id="spotify"
      >
        <img
          src="spotify-logo.png"
        />
      </div>
    </a>
    <a>
      <div
        className="music-logo"
        id="youtube"
      >
        <img
          src="youtube-logo.png"
        />
      </div>
    </a>
    <a>
      <div
        className="music-logo"
        id="facebook"
      >
        <img
          src="facebook-logo.png"
        />
      </div>
    </a>
    <a>
      <div
        className="music-logo"
        id="bandcamp"
      >
        <img
          src="bandcamp-logo.png"
        />
      </div>
    </a>
    <a>
      <div
        className="music-logo"
        id="website"
      >
        SITE
      </div>
    </a>
  </div>
</div>
`;
